### PR TITLE
Allow user to add groups to a container

### PR DESCRIPTION
**Pull request by bol**
Adds support for the --add-group parameter for container creation and inspection.
Solves #907

---
This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/950)

### CODE REVIEW

File: src/main/java/com/github/dockerjava/core/command/CreateContainerCmdImpl.java [hunk 1]
```diff
@@ -987,6 +993,13 @@ public CreateContainerCmd withHostConfig(HostConfig hostConfig) {
         return this;
     }
 
+    @Override
+    public CreateContainerCmd withGroupAdd(String[] groupAdd) {
+        checkNotNull(groupAdd, "groupAdd was not specified");
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/command/CreateContainerCmdImpl.java]: remove null check

File: src/main/java/com/github/dockerjava/api/model/HostConfig.java [hunk 2]
```diff
@@ -816,6 +818,23 @@ public HostConfig withPidsLimit(Long pidsLimit) {
         this.pidsLimit = pidsLimit;
         return this;
     }
+
+
+    /**
+     * @see #groupAdd
+     */
+    @CheckForNull
+    public String[] getGroupAdd() {
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/model/HostConfig.java]: use lists

**Comment by bol** [comment]
Updated this PR with changes from master. Build is currently broken because eclipse.org is sending a 302 with a reference to their error page on a missing dependency instead of a proper 404. Should build cleanly again when their repo repo is back in a sane state.

File: src/main/java/com/github/dockerjava/core/command/CreateContainerCmdImpl.java [hunk 3]
```diff
@@ -267,6 +267,12 @@ public Integer getCpuShares() {
         return hostConfig.getDnsSearch();
     }
 
+    @Override
+    @JsonIgnore
+    public List<String> getGroupAdd() {
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/command/CreateContainerCmdImpl.java]: don't proxy host changes, keep only getter/setters in hostconfig

Comment (bol) [reply src/main/java/com/github/dockerjava/core/command/CreateContainerCmdImpl.java]: Would that not make it awkward to use?
I saw several parts of hostconfig being proxied, is there a practice for which of them to expose?

**Pull request closed without merging**